entrypoint.go

```diff
@@ -67,17 +80,44 @@ func main() {
 	}
 	log.Printf("entrypoint: started OTel successfully")
 
-	// Wait for signals from Cloud Run. Signal the sub process appropriately
-	// after making relevant changes to the config and/or health signals.
-	// TODO(b/307317433): Consider having a timeout to shutdown the subprocess
-	// non-gracefully.
-	sig := <-signalChan
-	log.Printf("entrypoint: %s signal caught", sig)
-
-	collectorProcess.Signal(sig)
-	processState, err := collectorProcess.Wait()
+	refreshTicker := time.NewTicker(configRefreshInterval)
+	lastStat, err := os.Stat(userConfigFile)
 	if err != nil {
-		log.Fatalf(processState.String(), err)
+		log.Fatal(err)
+	}
+	for {
+		select {
+		case <-refreshTicker.C:
+			stat, err := os.Stat(userConfigFile)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			if stat.Size() != lastStat.Size() || stat.ModTime() != lastStat.ModTime() {
```

Comment:
A change in file size would imply a change in modification time, so we don't really need the size check here.

Reply:
I'm seeing the mod time not update when the secret is updating. I'm keeping the size check here for now too, but will contact the Cloud Run team to see why this is happening.